frontend/src/pages/modelRegistry/ModelRegistryCoreLoader.tsx

```diff
@@ -19,67 +20,72 @@ type ModelRegistryCoreLoaderProps = {
+    const { modelRegistries, preferredModelRegistry } = React.useContext(
+      ModelRegistrySelectorContext,
+    );
 
-  let renderStateProps: ApplicationPageRenderState & { children?: React.ReactNode };
-  if (modelRegistries.length === 0) {
-    renderStateProps = {
-      empty: true,
-      emptyStatePage: (
-        // TODO: Replace this with a component for empty registries once we have the designs
-        <EmptyModelRegistryState
-          title="No model registries found"
-          description="No model registries found in the cluster. Configure a new one before registering models."
-          primaryActionText="Configure model registry"
-          primaryActionOnClick={() => {
-            // TODO: Add primary action
-          }}
-        />
-      ),
-    };
-  } else if (modelRegistry) {
-    const foundModelRegistry = modelRegistries.find((mr) => mr.metadata.name === modelRegistry);
-    if (foundModelRegistry) {
-      // Render the content
-      return (
-        <ModelRegistryContextProvider modelRegistryName={modelRegistry}>
-          <Outlet />
-        </ModelRegistryContextProvider>
-      );
-    }
+    let renderStateProps: ApplicationPageRenderState & { children?: React.ReactNode };
+    if (modelRegistries.length === 0) {
+      renderStateProps = {
+        empty: true,
+        emptyStatePage: (
+          // TODO: Replace this with a component for empty registries once we have the designs
+          <EmptyModelRegistryState
+            title="No model registries found"
+            description="No model registries found in the cluster. Configure a new one before registering models."
+            primaryActionText="Configure model registry"
+            primaryActionOnClick={() => {
+              // TODO: Add primary action
+            }}
+          />
+        ),
+      };
+    } else if (modelRegistry) {
+      const foundModelRegistry = modelRegistries.find((mr) => mr.metadata.name === modelRegistry);
+      if (foundModelRegistry) {
+        // Render the content
+        return (
+          <ModelRegistryContextProvider modelRegistryName={modelRegistry}>
+            <Outlet />
+          </ModelRegistryContextProvider>
+        );
+      }
 
-    // They ended up on a non-valid project path
-    renderStateProps = {
-      empty: true,
-      emptyStatePage: <InvalidModelRegistry modelRegistry={modelRegistry} />,
-    };
-  } else {
-    // Redirect the namespace suffix into the URL
-    const redirectModelRegistry = preferredModelRegistry ?? modelRegistries[0];
-    return <Navigate to={getInvalidRedirectPath(redirectModelRegistry.metadata.name)} replace />;
-  }
+      // They ended up on a non-valid project path
+      renderStateProps = {
+        empty: true,
+        emptyStatePage: <InvalidModelRegistry modelRegistry={modelRegistry} />,
+      };
+    } else {
+      // Redirect the namespace suffix into the URL
+      const redirectModelRegistry = preferredModelRegistry ?? modelRegistries[0];
```

Comment:
Interesting to have a fallback here even though the same is done in the context.

Reply:
I wonder if that was done simply because the type can be undefined, which is necessary because there's no default value for the context until registries are loaded. Maybe the solution is to instead just not return the `<Navigate>` here if it's undefined (which as you say won't really happen)

Reply:
Oh i thought i deleted this comment. This is fine to keep because types require you to do something.